src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -11041,12 +11071,13 @@ private LambdaExpressionSyntax ParseLambdaExpression(SyntaxToken asyncToken)
             else
             {
                 var name = this.ParseIdentifierToken();
+                var exclamation = this.CurrentToken.Kind == SyntaxKind.ExclamationToken ? this.EatToken(SyntaxKind.ExclamationToken) : null;
```

Comment:
can be smiplified

Reply:
Perhaps a prototype comment for this and similar checks which can be resolved after you merge master into the feature branch.